### PR TITLE
Md huzaifa 1146

### PR DESCRIPTION
**Pull request by md-huzaifa**
Created the Book and Cart pages with functionlaity to add and remove the books to cart and check the total amount

### CODE REVIEW

**Review by atmeshwar-classic — CHANGES_REQUESTED** [review]
//UPDATE
Sorry my bad, didn't fetch it soon enough :)

Hi @md-huzaifa , Please finish the Cart page, it is incomplete.

File: package.json [hunk 1]
```diff
@@ -12,12 +12,17 @@
...
+    "react-router-dom": "^6.8.0",
+    "react-toastify": "^9.1.1"
   },
   "devDependencies": {
     "@types/react": "^18.0.26",
     "@types/react-dom": "^18.0.9",
     "@vitejs/plugin-react": "^3.0.0",
+    "autoprefixer": "^10.4.13",
+    "postcss": "^8.4.21",
+    "tailwindcss": "^3.2.4",
```
Comment (atmeshwar-classic) [inline package.json:25]: Nice touch by adding Tailwind 😄

File: src/App.tsx [hunk 2]
```diff
@@ -1,34 +1,20 @@
...
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </div>
+    <Router>
```
Comment (atmeshwar-classic) [inline src/App.tsx]: Please extract the routes into its own separate file and import them here.
For eg: src/routes/routes.ts

We should maintain the structure of the app regardless of its size.

File: src/components/NavBar/NavBar.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const NavBar: React.FC<NavBarProps> = ({}) => {
+  return (
+    <nav className="flex items-center justify-around flex-wrap bg-teal-500 p-6">
+      <ul className="flex justify-evenly">
+        <li className="mx-10">
+          <Link to="/books">Books</Link>
+        </li>
+        <li className="mx-10">
+          <Link to="/">Home</Link>
+        </li>
```
Comment (atmeshwar-classic) [inline src/components/NavBar/NavBar.tsx]: No need for Home link as the Books page is the default one.

File: src/domain/books/BooksPage.tsx [hunk 4]
```diff
@@ -1,7 +1,42 @@
...
+import type { Book } from "./types";
+
 export const BooksPage = (): JSX.Element => {
+  const { books, loading } = useSelector(booksSelector);
+
   return (
     <div>
-      Books Page
+      <h1 className="ml-10 mt-10">Browse our book collection:</h1>
+      <ToastContainer draggable />
```
Comment (atmeshwar-classic) [inline src/domain/books/BooksPage.tsx]: ToastContainer should be at the root of our app in main.tsx.

File: src/domain/cart/CartPage.tsx [hunk 5]
```diff
@@ -1,5 +1,83 @@
...
+  const cartItems:CartProducts[] = useSelector(cartSelector);
+  const total = cartItems.reduce((a, c) => (a += c.price * c.quantity), 0);
+
   return (
-    <div>Cart Page</div>
-  )
-}
\ No newline at end of file
+    <div className="container mx-auto mt-5 w-8/12">
+      <ToastContainer />
```
Comment (atmeshwar-classic) [inline src/domain/cart/CartPage.tsx]: See above.

**Review by atmeshwar-classic — CHANGES_REQUESTED** [review]
Hi @md-huzaifa Please check the comments I have made.

Also, whenever I click on add to cart button the url adds a '#' in the end. Please look into it as well.

File: package.json (hunk 1 above)
Comment (md-huzaifa) [reply package.json:25]: Thank you

File: src/routes/routes.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { Route } from "react-router-dom";
+import { BooksPage } from "../domain/books/BooksPage";
+import { CartPage } from "../domain/cart/CartPage";
+
+const createRoutes = () => {
+  return (
+    <>
+      <Route path="/" element={<BooksPage />} />
+      <Route path="/books" element={<BooksPage />} />
```
Comment (atmeshwar-classic) [inline src/routes/routes.tsx]: Not needed

File: src/routes/routes.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Route } from "react-router-dom";
+import { BooksPage } from "../domain/books/BooksPage";
+import { CartPage } from "../domain/cart/CartPage";
+
+const createRoutes = () => {
+  return (
+    <>
```
Comment (atmeshwar-classic) [inline src/routes/routes.tsx]: You can use Routes element here instead of in App.tsx

File: src/App.tsx [hunk 8]
```diff
@@ -1,20 +1,16 @@
-import { BrowserRouter as Router, Link, Route, Routes } from "react-router-dom";
+import { Routes } from "react-router-dom";
 import "./App.css";
 import NavBar from "./components/NavBar/NavBar";
-import { BooksPage } from "./domain/books/BooksPage";
-import { CartPage } from "./domain/cart/CartPage";
+import createRoutes from "./routes/routes";
 
 function App() {
+  const routes = createRoutes();
```
Comment (atmeshwar-classic) [inline src/App.tsx]: Just import Routes like any other React component, no need to do this.

File: src/routes/routes.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { Route } from "react-router-dom";
+import { BooksPage } from "../domain/books/BooksPage";
+import { CartPage } from "../domain/cart/CartPage";
+
+const createRoutes = () => {
```
Comment (atmeshwar-classic) [inline src/routes/routes.tsx]: Just rename it to Routes or Router whatever you like and import it in App like a react component

**Comment by atmeshwar-classic** [comment]
I am closing this PR as the assessment is complete.
Good job @md-huzaifa . Keep it up!

**Comment by md-huzaifa** [comment]
Thank you @atmeshwar-classic

**Pull request closed without merging**